.github/workflows/deb.yml

```diff
@@ -0,0 +1,60 @@
+name: Build Debian Packages
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version to release'
+        required: true
+      upload-artifacts:
+        description: 'Upload artifacts to PackageCloud'
+        required: false
+        default: false
+      
+jobs:
+  build-deb:
+    runs-on: ${{ matrix.runner }}
+    container:
+      image: timescaledev/rust-pgx:latest
+    strategy:
+      matrix:
+        pgversion: [12, 13, 14]
+        runner:
+          - ubuntu-18.04
+          - ubuntu-20.04
+          - ubuntu-22.04
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.inputs.version }}
+
+      - name: Cache cargo directories
+        uses: actions/cache@v2
+        with:
+          path: |
+            /usr/local/cargo/registry
+            /usr/local/cargo/git
+          key: ${{ runner.os }}-test-pg${{ matrix.pgversion }}-cargo-1.8
+          restore-keys: ${{ runner.os }}-test-pg${{ matrix.pgversion }}-cargo-1.5
+
+      - name: Cache cargo target dir
+        uses: actions/cache@v2
+        with:
+          path: target
+          key: ${{ runner.os }}-test-pg${{ matrix.pgversion }}-target-1.8
+          restore-keys: ${{ runner.os }}-test-pg${{ matrix.pgversion }}-target-1.5
+
+      - name: Build Debian Package
+        id: package
```

Comment:
This id is not needed. I intended to use it to output the debian files built so that they can be referenced in the following action, but that is not done here.